### ISSUE TITLE
NullPointerException looking for CppCheck report

### ISSUE THREAD

**Issue by tyrel** [SonarOpenCommunity/sonar-cxx#766]
I'm getting a NPE when I attempt to run analysis on my project, with sonar-cxx 0.9.3.

```
INFO: Sensor CxxCoverageSensor (done) | time=1526ms
INFO: Sensor CxxCppCheckSensor
DEBUG: Using pattern 'null' to find reports
INFO: Analysis report generated in /home/gitlab-runner/builds/580bbdff/0/hidden/path/src/.sonar/batch-report
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 15.928s
INFO: Final Memory: 41M/110M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarQube Scanner execution
org.sonar.api.utils.SonarException: Cannot feed the data into sonar, details: 'java.lang.NullPointerException'
        at org.sonar.plugins.cxx.utils.CxxReportSensor.analyse(CxxReportSensor.java:134)
        at org.sonar.batch.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:58)
        at org.sonar.batch.phases.SensorsExecutor.execute(SensorsExecutor.java:50)

.... snip ....

Caused by: java.lang.NullPointerException
        at org.apache.tools.ant.DirectoryScanner.setIncludes(DirectoryScanner.java:569)
        at org.sonar.plugins.cxx.utils.CxxReportSensor.getReports(CxxReportSensor.java:166)
        at org.sonar.plugins.cxx.utils.CxxReportSensor.analyse(CxxReportSensor.java:104)
        ... 31 more
```
I _am_ setting the report path in my properties file:
```
sonar.cxx.cppcheck.reportPath=../out/doc/cppcheck/*.xml
```

It looks like there's been a lot of refactoring going on in your repository here so the code that caused this may no longer exist, but maybe this problem is still there. In the code that went into 0.9.3, I'm not sure what would cause reportPath to be null, because (1) I'm setting it in my config, and (2) it's getting set to the default if it is null.

**Comment by jmecosta** [comment]
Is there something preventing you from using the last 0.9.4, so you can verify if its still broken?

**Comment by tyrel** [comment]
Okay, I'm not sure how I managed to get 0.9.3. I was certain I downloaded latest a couple days ago when I got it. I will try 0.9.4, but by now I've restructured my project to handle this better so I probably won't run into it. It seems to have something to do with the relative path in `reportPath` with the `..` in it. I tried setting it to just `out/cppcheck/*.xml` (and have CppCheck put its output there) and it then worked fine.

**Comment by tyrel** [comment]
I still get this with 0.9.4, any time the reportPath contains a `..`

**Comment by jmecosta** [comment]
you can try last https://ci.appveyor.com/project/SonarOpenCommunity/sonar-cxx/branch/master/artifacts although not sure if the .. will work. this use case was never supported.

**Comment by guwirth** [comment]
@tyrel could you provide a sample path please?
* is the issue in the settings.conf or in a report file?
* is the path with ´..´ relative or absolute?

**Comment by tyrel** [comment]
@guwirth The path is specified in the sonar.project.properties as something like `../out/doc/cppcheck/*.xml` when it fails, so it's a relative path. You can see this example in the OP.

**Comment by mjdetullio** [comment]
This should be fixed now that #777 is merged

**Comment by mjdetullio** [comment]
I lied... there's actually still a NPE with my solution.  `FilenameUtils.normalize(String)` will return null for invalid paths or relative paths that have backticks outside the directory tree known by the relative path.
